src/lib/core/overlay/position/viewport-ruler.ts

```diff
@@ -51,6 +54,12 @@ export class ViewportRuler {
    * @param documentRect
    */
   getViewportScrollPosition(documentRect = this._documentRect) {
+    // Cache the document bounding rect so that we don't recompute it for multiple calls.
+    if (!documentRect) {
+      this._cacheViewportGeometry();
```

Comment:
This could be a little shorter if we made `_cacheViewportGeometry` return the new `documentRect` and then changed the method definition to:

```ts
getViewportScrollPosition(documentRect = this._cacheViewportGeometry()) {
  // continue as usual
}
```